src/colorizer/CanvasWithOverlay.ts

```diff
@@ -72,33 +128,58 @@ export default class CanvasWithOverlay extends ColorizeCanvas {
   private canvas: HTMLCanvasElement;
   private ctx: CanvasRenderingContext2D;
 
+  private collection: Collection | null;
+  private datasetKey: string | null;
+
   private scaleBarOptions: ScaleBarOptions;
   private timestampOptions: TimestampOptions;
-  private backgroundOptions: OverlayFillOptions;
+  private overlayBoxOptions: OverlayBoxOptions;
+  private legendOptions: LegendOptions;
+  private headerOptions: HeaderOptions;
+  private footerOptions: FooterOptions;
   private canvasWidth: number;
   private canvasHeight: number;
-
-  constructor(
-    scaleBarOptions: ScaleBarOptions = defaultScaleBarOptions,
-    timestampOptions: TimestampOptions = defaultTimestampOptions,
-    overlayOptions: OverlayFillOptions = defaultBackgroundOptions
-  ) {
+  private headerSize: Vector2;
+  private footerSize: Vector2;
+
+  private isExporting: boolean;
+
+  constructor(options?: {
+    scaleBar?: ScaleBarOptions;
+    timestamp?: TimestampOptions;
+    background?: OverlayBoxOptions;
+    legend?: LegendOptions;
+    header?: HeaderOptions;
+    footer?: FooterOptions;
+  }) {
     super();
 
     this.canvas = document.createElement("canvas");
     this.canvas.style.display = "block";
 
-    this.scaleBarOptions = scaleBarOptions;
-    this.timestampOptions = timestampOptions;
-    this.backgroundOptions = overlayOptions;
+    this.collection = null;
+    this.datasetKey = null;
+
+    this.scaleBarOptions = options?.scaleBar || defaultScaleBarOptions;
+    this.timestampOptions = options?.timestamp || defaultTimestampOptions;
+    this.overlayBoxOptions = options?.background || defaultOverlayBoxOptions;
+    this.legendOptions = options?.legend || defaultLegendOptions;
+    this.headerOptions = options?.header || defaultHeaderOptions;
+    this.footerOptions = options?.footer || defaultFooterOptions;
```

Comment:
If I need to add more elements in the future I may just condense them all into one shared options object. TBD!